lib/cli/src/Cardano/CLI.hs

```diff
@@ -685,16 +687,17 @@ cmdWalletGetUtxoStatistics mkClient =
 -- | cardano-wallet transaction
 cmdTransaction
     :: ToJSON wallet
-    => TransactionClient
+    => Bool -- ^ Enable Shelley transaction features
```

Comment:
May I suggest a sum-type: `Era = Byron | Shelley (Goguen Voltaire)` and a possible `isShelley :: Era -> Bool` derived from it? This would be less cryptic at the call-site when reading: `cmdTransaction True client ...`

Reply:
OK I have added a sum type but kept it hidden in the CLI module.